aws/components/apigateway/state.ftl

```diff
@@ -365,6 +365,7 @@ created in either case.
                 },
                 "apideploy" : {
                     "Id" : formatResourceId(AWS_APIGATEWAY_DEPLOY_RESOURCE_TYPE, core.Id, getCLORunId()),
+                    "Name": formatName(apiName,"deploy"),
```

Comment:
Names don't need to include the type. The actual resource sets that context anyway

Reply:
I was assuming the names needed to be unique for the AWS component, so if one hamlet component spawned multiple AWS components, we needed to differentiate between them. 

Do I have that backwards and it should be the hamlet component name (or something else entirely)?

Reply:
Ids and Names are generated to be unique as they are formed on the occurrence, so as you split out new occurences the names are always going to be unique across a component. 

Names within AWS are usually for tags which are just metadata and don't need to be unique. If they do need to be unique they only need to be unique within the particular resource .

If you used the same resource across different components then the names would still be unique because all occurrences are unique

Reply:
Changed, rebased and commit message changed to conventional format